### PR TITLE
Packet GET routes for Conditional

### LINKED ISSUE #176: Endpoint to get packet score
Conditional should be able to pull packet scores automatically from packet. I want to create 2 endpoints for getting this information:

- [x] `/v1/packets/{username}` - Which collects all packets that person has completed and their start and end dates and id
- [x] `/v1/packet/{packetId}` - Displays the scores of the packet in question

Potential Route:
- [x] `/v1/packet/{username}/newest` - Which collects the newest packet that person has completed and their start and end dates and scores (For easy use with services like conditional)

Implementation conditional side should be described in an issue on that repo

### PR DESCRIPTION
**Pull request by mxmeinhold**
Resolves #176

### CODE REVIEW

File: packet/routes/api.py [hunk 1]
```diff
@@ -7,10 +7,61 @@
...
+
+
+@app.route("/api/v1/packets/<username>/newest", methods=["get"])
+@packet_auth
+@before_request
+def get_newest_packet_by_user(username: str, info) -> dict:
+    """
+    Return a user's newest packet
+    """
+    frosh = Freshman.by_username(username)
```
[the remark this answers is not in the document]
Comment (mxmeinhold) [reply packet/routes/api.py]: It does for queries, I've yet to find it for one to many relationships. The docs don't seem to mention any operations on them, just arguments for creating them. Everything I've touched with them they seem to act like lists.

File: packet/routes/api.py [hunk 2]
```diff
@@ -7,14 +7,62 @@
 from packet.context_processors import get_rit_name
 from packet.mail import send_report_mail
 from packet.utils import before_request, packet_auth, notify_slack
-from packet.models import Packet, MiscSignature, NotificationSubscription
+from packet.models import Packet, MiscSignature, NotificationSubscription, Freshman
 from packet.notifications import packet_signed_notification, packet_100_percent_notification
 
 
-@app.route("/api/v1/sign/<packet_id>/", methods=["POST"])
+@app.route("/api/v1/packets/<username>", methods=["get"])
```
Comment (devinmatte) [inline packet/routes/api.py]: Please follow the above format and make this `"GET"`

File: packet/routes/api.py [hunk 3]
```diff
@@ -7,14 +7,62 @@
...
+    """
+    frosh = Freshman.by_username(username)
+
+    return {packet.id: {
+        'start': packet.start,
+        'end': packet.end,
+        } for packet in frosh.packets}
+
+
+@app.route("/api/v1/packets/<username>/newest", methods=["get"])
```
Comment (devinmatte) [inline packet/routes/api.py]: Same here. `GET`

File: packet/routes/api.py [hunk 4]
```diff
@@ -7,14 +7,62 @@
...
+            packet.id: {
+                'start': packet.start,
+                'end': packet.end,
+                'required': packet.signatures_required().as_dict(),
+                'received': packet.signatures_received().as_dict(),
+                }
+            }
+
+
+@app.route("/api/v1/packet/<packet_id>", methods=["get"])
```
Comment (devinmatte) [inline packet/routes/api.py]: Same as above `GET`

File: packet/routes/api.py [hunk 5]
```diff
@@ -7,14 +7,62 @@
...
+    """
+    frosh = Freshman.by_username(username)
+
+    packet = frosh.packets[-1]
+
+    return {
+            packet.id: {
+                'start': packet.start,
+                'end': packet.end,
+                'required': packet.signatures_required().as_dict(),
```
Comment (JoelEager) [inline packet/routes/api.py]: Isn't this broken now that `.as_dict()` is gone?

**Comment by mxmeinhold** [comment]
If this looks alright, I'd like to squash it before merging.

File: packet/routes/api.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (devinmatte) [reply packet/routes/api.py]: It's not a huge deal, but for data reduction you could order desc and then do `.first()` or `limit(1)`

**Pull request merged**